spot-electron/src/bt-beacon/daemon/darwin/bluetoothd.js

```diff
@@ -0,0 +1,159 @@
+const XpcConnection = require('xpc-connect');
+
+const { logger } = require('../../../logger');
+
+const DEVICE_STATES = [ 'Unknown', 'Resetting', 'Unsupported', 'Unauthorized', 'Powered Off', 'Powered On' ];
+
+/**
+ * Implements a platform specific daemon to interact with the BLE hardware in the computer/device.
+ */
+class BluetoothD {
+    advertisingStatusUpdatedPromise;
+
+    /**
+     * Instantiates a new instance.
+     */
+    constructor() {
+        this.xpcConnection = new XpcConnection('com.apple.bluetoothd');
+
+        this.xpcConnection.on('error', message => {
+            logger.error('XpcError', message);
+        });
+
+        this.xpcConnection.on('event', event => {
+            const messageArguments = event.kCBMsgArgs;
+
+            switch (event.kCBMsgId) {
+            case 4:
+                // Device status update
+                this.setStatus(messageArguments.kCBMsgArgState);
+                break;
+            case 28:
+                // Beacon activation result
+                this.setAdvertisingStatus(!messageArguments.kCBMsgArgResult);
+                break;
+            case 29:
+                // Beacon deactivation result
+                this.setAdvertisingStatus(Boolean(messageArguments.kCBMsgArgResult));
+                break;
+            case 38:
+                // Device ready
+                this.readyPromise && this.readyPromise();
+                break;
+            case 67:
+                // Connection request. We don't need this.
+                break;
+            default:
+                logger.warn('Unknown XpcEvent', event);
+            }
+        });
+    }
+
+    /**
+     * Connects to the BT device (opens XPC channel).
+     *
+     * @returns {Promise<void>}
+     */
+    connect() {
+        return new Promise(resolve => {
+            this.readyPromise = resolve;
+            this.xpcConnection.setup();
+            this.sendMessage(1, {
```

Comment:
Why is the message id hardcoded as 1?

Reply:
Message IDs are not sequence in case of XPC. Those actually denote the command we want XPC to execute. Message ID 1 is the command to power on the BT device

Reply:
I wonder then if a rename is necessary to denote these are intentionally constant. But maybe converting the magic numbers to constants would resolve that.